lib/language/language_constants.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:we_find/model/i18n_string.dart';
+import 'package:we_find/providers/lang_provider.dart';
+import 'package:we_find/screens/advaned_search_screen.dart';
+
+class LanguageConstatnts {
+  final BuildContext _context;
+  LanguageConstatnts(this._context);
+
+  Lang get lang => Provider.of<LangProvider>(_context).currentLang;
+  BuildContext get context => _context;
+
+  // missing data
+  String get NotFound => lang == Lang.EN ? "Not Found" : "Nicht gefunden";
+  String get Unknown => lang == Lang.EN ? "Unknown" : "Nicht bekannt";
+  String get UnknownSemester =>
+      lang == Lang.EN ? "Unknown Semester" : "Semester nicht bekannt";
+  String get UnknownDepartment =>
+      lang == Lang.EN ? "Unknown Department" : "Institut nicht bekannt";
+  String get TitleNotFound =>
+      lang == Lang.EN ? "Title not found" : "Titel nicht gefunden";
+
+  // advanced search
+  String get FuzzySearch => lang == Lang.EN ? "Fuzzy Search" : "Fuzzy Suche";
+  String get CourseWithStream =>
+      lang == Lang.EN ? "Course with Stream" : "Kurs mit Stream";
+  String get OnlyCurrentTerm =>
+      lang == Lang.EN ? "Only current term" : "Nur dieses Semester";
+  String get SearchByTerm =>
+      lang == Lang.EN ? "Search by Term" : "Suche im Semester";
+  String get SearchByTermRange =>
+      lang == Lang.EN ? "Search by Term Range" : "Suchen in Semesterintervall";
+  String get FromTerm => lang == Lang.EN ? "From Term" : "Von Semester";
```

Comment:
Change these to "Semester von" / "Semester bis" imo, seen here: https://ufind.univie.ac.at/de/help.html